internal/discovery/client.go

```diff
@@ -145,25 +144,25 @@ func (m *discoveryClient) pollClusterInfo() {
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
 	defer cancel()
 	response, err := m.dkvClient.GetClusterInfo(ctx, &serverpb.GetClusterInfoRequest{})
-	if (err != nil) {
+	if err != nil {
 		m.logger.Error("Unable to poll cluster info", zap.Error(err))
 	} else {
 		m.clusterInfo = response.GetRegionInfos()
 	}
 }
 
 func (m *discoveryClient) GetClusterStatus(database string, vBucket string) ([]*serverpb.RegionInfo, error) {
-	if (m.clusterInfo == nil) {
+	if m.clusterInfo == nil {
 		// When called before cluster info is initialised
 		m.pollClusterInfo()
-		if (m.clusterInfo == nil) {
+		if m.clusterInfo == nil {
 			// in case cluster info initialise fails
 			return nil, fmt.Errorf("Unable to get cluster info")
 		}
 	}
-	regions := []*serverpb.RegionInfo{}
+	var regions []*serverpb.RegionInfo
```

Comment:
@kingster initing to empty array is better than nil right ? if no node is active for required filter, response should be empty array rather than nil.

Reply:
Nil and empty slices are the same in golang

https://stackoverflow.com/questions/29164375/correct-way-to-initialize-empty-slice